library.gradle

```diff
@@ -43,9 +51,21 @@ dependencies {
     testImplementation 'org.robolectric:robolectric:4.6.1'
     testImplementation 'com.squareup.okhttp3:mockwebserver:4.2.0'
     testImplementation 'org.mockito:mockito-core:3.0.0'
-    testImplementation "com.android.billingclient:billing:$billingVersion"
+    testNewBillingImplementation "com.android.billingclient:billing:$billingVersion"
+    testOldBillingImplementation "com.android.billingclient:billing:$oldBillingVersion"
     testImplementation 'io.mockk:mockk:1.10.0'
     testImplementation 'org.assertj:assertj-core:3.13.2'
 }
 
+def artifactId = project.property("POM_ARTIFACT_ID")
+def publishVariant = project.property("PUBLISH_VARIANT")
+if (publishVariant == "oldBillingRelease") {
+    project.ext.POM_ARTIFACT_ID = artifactId + "-unityiap"
```

Comment:
I am honestly not sure if we should add `-unityiap` or `-billing3` to the versions. I left it like this to get some feedback:

- `-unityiap` has the advantage of making it clear that this is created mainly for unity
- `-billing3` has the drawback that this version will also have an older version of the Amazon SDK (the one that Unity IAP uses).

Reply:
Btw, there will be another PR adding the older version of the Amazon SDK. But that one should be easier because nothing has changed so the logic is the same.

Reply:
this would still be helpful for folks in observer mode who're still using v3, though, right? 

I'm still not very clear on the implications of `-unityiap` vs `-billing3`, could you elaborate?

Reply:
Sorry I wasn't very clear explaining this. Unity IAP uses Amazon v2 and BillingClient v3. The latest version of purchases-android uses Amazon v3 and BillingClient v4. 

So we can have 2 flavors, one with Amazon v3 + BillingClient 4 and another one with Amazon v2 and BillingClient 3. Or we can have 4 flavors for all combinations:  Amazon v3 + BillingClient 4, Amazon v3 + BillingClient 3, Amazon v2 + BillingClient 4 and Amazon v2 + BillingClient 3.

I think the main purpose of these flavors are to be able to support Unity IAP with the latest version of purchases-android, it's true that it can be helpful for folks in observer mode using v3, but they are most likely able to upgrade to v4, and we should push for that instead of helping them stay in v4.

That's why I think we should name the flavor maybe -unityiap, meaning that it will have billing3 and amazon2. 

I hope this is more clear.

Reply:
@aboedo any thoughts?

Reply:
ohh I see. 
I do agree that we should just cover the most basic use case, we can always add more flavors if / when they become absolutely necessary. 
I think just `-unityiap` is good, since it's by far the most common use case, and it makes it abundantly clear that it's the one you should use if you use `unityIAP`